City-of-Helsinki/drupal-helfi-rekry
# UHF-5286: Schedule and hide job listings

# Publish and unpublish job listings [UHF-5286](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5286)

## What was done
* Imports publish and unpublish times during job listing migration.
* Redirects anonymous users to another page when trying to view unpublished job listing.
* Added boolean field to prevent automatic publishing.

## How to install
* Make sure your instance is up and running.
* Get the changes:
  * `git pull origin UHF-5286-hide-job-listings`
* Run `make shell` and
  * `drush cim`
  * `drush cr`

## How to test
* Import the job listings:
  * `drush migrate:import helfi_rekry_jobs:all`
* Check that imported job listings have scheduled publishing and unpublishing times (if the job listing has those at the API).
* Run `drush cron` to publish and unpublish job listings according the publishing and unpublishing times. Check that the jobs are published and unpublished accordingly.
* Create a new basic page. Add some text that informs the user that the job listing is no longer available.
* Go to `/admin/config/services/rekry-content`. You can see that there is prefilled field with value `/job-listing-removed`.
* Go to `/admin/config/search/redirect`. Add a new redirect from `/job-listing-removed` to your new info page.
* Find a job listing that is unpublised. Check that a logged in user can see it.
  * Next, try to access the page with anonymous user. You should be redirected to your info page.
* Test the job listing's prevent publishing option:
  * Unpublish some job listing if it's published.
  * Select the prevent publishing option.
  * Set the publish date so that the job listing is going to be published after couple of minutes.
  * After that time, run `drush cron` so that the job listing would be published if the prevent publishing option wouldn't be selected.
  * Make sure the job listing is not published.
  * You can now unselect the option, run the cron again and see that the job listing is now published.

## Designers review

One of the checkboxes below needs to be checked (`[x]`)
